engine/src/main/java/io/camunda/zeebe/engine/processing/EngineProcessors.java

```diff
@@ -240,4 +246,15 @@ private static void addMessageProcessors(
         subscriptionCommandSender,
         writers);
   }
+
+  private record VariableStateEvaluationContextLookup(VariablesLookup lookup)
+      implements EvaluationContextLookup {
+
+    @Override
+    public EvaluationContext getContext(final long scopeKey) {
+      ensureGreaterThan("variable scope key", scopeKey, 0);
+
+      return (name) -> lookup.getVariable(scopeKey, BufferUtil.wrapString(name));
+    }
+  }
```

Comment:
Solved in the opposite way. Instead of making it a standalone record, I made it an anonymous lambda. WDYT??

Reply:
IMO, this logic is just not something that the EngineProcessors class should care about. I'd prefer a standalone class/record in the `io.camunda.zeebe.engine.variable` package. One could also argue that the nested lambdas break the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) slightly because it is not so common in Java. 

Personally, I don't really like it, but if you want to keep it that's fine.